File: RentalApp_FASTAPI/shared/constants/deposit_status.py
# shared/constants/deposit_status.py

from enum import Enum


class DepositStatus(str, Enum):
    """Статусы залога в жизненном цикле аренды."""
    HELD = "held"                          # Залог удерживается (взят при оформлении аренды)
    REFUNDED = "refunded"                  # Залог полностью возвращен клиенту
    PARTIALLY_RETAINED = "partially_retained"  # Залог частично удержан (например, компенсация мелких дефектов)
    RETAINED_FOR_DAMAGE = "retained_for_damage" # Залог полностью удержан за ущерб / утрату


DEPOSIT_STATUS_LABELS = {
    DepositStatus.HELD: "Удерживается",
    DepositStatus.REFUNDED: "Возвращен клиенту",
    DepositStatus.PARTIALLY_RETAINED: "Частично удержан",
    DepositStatus.RETAINED_FOR_DAMAGE: "Удержан за ущерб",
}

File: rental-app-main/src/constants/depositStatus.ts
// src/constants/depositStatus.ts

export type DepositStatus = 'held' | 'refunded' | 'partially_retained' | 'retained_for_damage';

export const DEPOSIT_STATUS_LABELS: Record<DepositStatus, string> = {
  held: 'Удерживается',
  refunded: 'Возвращен клиенту',
  partially_retained: 'Частично удержан',
  retained_for_damage: 'Удержан за ущерб',
};

export const DEPOSIT_ACTION_OPTIONS = [
  { value: 'refund', label: 'Вернуть залог клиенту' },
  { value: 'retain', label: 'Удержать залог полностью' },
  { value: 'partial_retain', label: 'Удержать частично' },
] as const;
